src/scf/mixing.jl

```diff
@@ -6,22 +6,30 @@
 
 # The interface is `mix(m, basis, ρin, ρout) -> ρnext`
 
+using LinearMaps
+using IterativeSolvers
+
+@doc raw"""
+Kerker mixing: ``J^{-1} ≈ \frac{α G^2}{(k_F^2 + G^2}``
+where ``k_F`` is the Thomas-Fermi screening constant.
+
+Notes:
+  - Abinit calls ``1/k_F`` the dielectric screening length (parameter *dielng*)
 """
-Kerker mixing: J^-1 ≈ α*G^2/(G0^2 + G^2)
-"""
-struct KerkerMixing{T <: Real}
-    α::T
-    G0::T
+struct KerkerMixing
+    α
+    kF
 end
-# Default parameters suggested by Kresse, Furthmüller 1996 (α=0.8, G0=1.5 Ǎ^{-1})
+# Default parameters suggested by Kresse, Furthmüller 1996 (α=0.8, kF=1.5 Ǎ^{-1})
 # DOI 10.1103/PhysRevB.54.11169
-KerkerMixing() = KerkerMixing(0.8, 0.8)
-function mix(m::KerkerMixing, basis, ρin::RealFourierArray, ρout::RealFourierArray)
+KerkerMixing(;α=0.8, kF=0.8) = KerkerMixing(α, kF)
+function mix(params::KerkerMixing; basis, ρin::RealFourierArray, ρout::RealFourierArray, kwargs...)
+    T = eltype(basis)
     Gsq = [sum(abs2, basis.model.recip_lattice * G)
            for G in G_vectors(basis)]
     ρin = ρin.fourier
     ρout = ρout.fourier
-    ρnext = @. ρin + m.α * (ρout - ρin) * Gsq / (m.G0^2 + Gsq)
+    ρnext = @. ρin + T(params.α) * (ρout - ρin) * Gsq / (T(params.kF)^2 + Gsq)
```

Comment:
I agree with that, but for the moment I don't see a simple solution but to annotate the types. Like if it is an RFA, then we annotate, otherwise not. I'll leave it for now as is.